core/tracepoint/defined_class_spec.rb

```diff
@@ -0,0 +1,48 @@
+require File.expand_path('../../../spec_helper', __FILE__)
+
+ruby_version_is '2.0' do
+  describe 'TracePoint#defined_class' do
+    module A; def bar; end; end
+
+    class B
+      include A
+      def foo; end;
+    end
+
+    class C < B
+      def initialize
+        super
+      end
+
+      def foo
+        super
+      end
+
+      def bar
+        super
+      end
+    end
+
+    it 'returns class or module of the method being called' do
+      class_name = nil
```

Comment:
This test is much improved :-)
I'd rename this variable `last_class_name` or something to make it clear that it will hold the class name of the last call.

Reply:
yes, `last_class_name` is good. I have renamed the variable & updated the PR.